Added user endpoint for the enrollments API and user filtering on enrollments endpoint (take 2)

johnbaldwin opened the pull request:
This PR replaces this PR: https://github.com/appsembler/edx-platform/pull/418

* This is the initial version of the user data API to support the enrollment API.
* Added UserIndexViewSet to provide user information via the Tahoe API
* Includes tests for the user viewset
* Updated enrollment test to support user filtering

I followed @OmarIthawi 's suggestion to just add `?user=<user id>` approach. However to do this with the course, we'd need to filter courses on the site, which would require passing the site to the filter, which requires investigating to make it work. Might be really simple, but I'd still have to investigate it.

So instead, I added `?user_id=<user-id>` and `?username=<username>` to the `/tahoe/api/v1/enrollments/` endpoint


This PR addresses the following comments from PR #418 

* https://github.com/appsembler/edx-platform/pull/418#pullrequestreview-259023694

which suggests using `user_id` instead of `user` in the `sites.get_user_ids_for_site` method

* https://github.com/appsembler/edx-platform/pull/418#discussion_r301193738
which uses `self.my_site_users` to validate the expected response set

Thanks @OmarIthawi !!!

OmarIthawi commented on openedx/core/djangoapps/appsembler/api/tests/test_enrollment_api.py:
```diff
@@ -23,7 +23,11 @@
 
 from organizations.models import UserOrganizationMapping
-from openedx.core.djangoapps.appsembler.api.sites import get_enrollments_for_site
+from openedx.core.djangoapps.appsembler.api.sites import (
```
Please remove the commented lines.

johnbaldwin replied:
Will do. Thanks for catching that

johnbaldwin commented:
@OmarIthawi Thanks for reviewing this so quickly!

johnbaldwin commented:
@OmarIthawi I addressed your PR comment and squashed all the commits except for the site packages update that includes the user getter methods and tests. I may want to have that separate to cherry pick for the api key user getter support. Hoping that it helps reduce some of the rebasing merge conflict
